### PR TITLE
feat: change account creation flow to ask sync database first

### LINKED ISSUE #1242: Clarify local password requirement when setting up 2nd, 3rd, ... machines for Premium syncing
As per #1187, it is currently a little too easy to miss the fact that when setting up more than one machine with Premium syncing between them, the local password needs to be the same on each since it's (currently) used as the local storage encryption key.

When creating a new account, there is an `Enable Premium` checkbox, with a tooltip next to it which warns that the password needs to match the one from the original machine, but this is way too easily missed.  I suggest that this warning is moved to be immediately under the password field.

Another reason for moving this is the user might not even notice the `Enable Premium` checkbox, and assume (correctly) that they can set up Premium later, after creating the account (which they can, in the API Keys panel).

See also #1239.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #1242 

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.
- [ ] Change account creation flow to ask sync database first
image

### CODE REVIEW

File: frontend/app/src/locales/en.json [hunk 1]
```diff
@@ -279,14 +279,23 @@
   },
   "create_account": {
     "title": "Create Account",
+    "sync_database": {
+      "title": "Sync Database",
+      "sync_question": "Do you want to sync Rotki database from another device? (This feature require premium credentials)",
```
Comment (kelsos) [inline frontend/app/src/locales/en.json:284]:
```suggestion
      "sync_question": "Do you want to sync your Rotki database from another device? (This feature requires premium credentials)",
```

File: frontend/app/src/locales/en.json [hunk 2]
```diff
@@ -279,14 +279,23 @@
...
     "select_credentials": {
       "title": "Select Credentials",
       "label_username": "Username",
       "label_password": "Password",
       "label_password_repeat": "Repeat Password",
       "label_password_backup_reminder": "rotki saves all user data locally. I understand that if I lose my password, I lose access to my data. I have created a backup of the password.",
-      "button_cancel": "Cancel",
+      "button_back": "Back",
       "button_continue": "Continue",
+      "password_sync_requirement": "Ensure that the account uses the same password as the database originally backed up. You can't access the database if you input different password.",
```
Comment (kelsos) [inline frontend/app/src/locales/en.json:298]:
```suggestion
      "password_sync_requirement": "Ensure that the account uses the same password as the database originally backed up. You can't access the database if you input a different password.",
```

**Review by kelsos — COMMENTED** [review]
I like the ideas here but maybe we could change the approach a bit.

In the first step instead of asking directly for the DB sync, we could instead ask the user if they want to set up their new local account with premium.

Then, only after they select the premium setup, which also allows them to insert the keys, we should ask them if they want to also restore the database from sync.

What do you think?

**Comment by lukicenturi** [comment]
> I like the ideas here but maybe we could change the approach a bit.
> 
> In the first step instead of asking directly for the DB sync, we could instead ask the user if they want to set up their new local account with premium.
> 
> Then, only after they select the premium setup, which also allows them to insert the keys, we should ask them if they want to also restore the database from sync.
> 
> What do you think?

Yeah, I think it would be better

**Comment by lukicenturi** [comment]
@kelsos updated

**Review by kelsos — APPROVED** [review]
lgtm, @lukicenturi we also need a change log entry here

**Comment by kelsos** [comment]
Something more, some of these PRs might need to update the [usage guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) since they change parts of the UI.

For the screenshots I usually use a custom window size in the dev tools:
image

**Comment by lukicenturi** [comment]
> lgtm, @lukicenturi we also need a change log entry here

change log and user guide are updated @kelsos

**Comment by kelsos** [comment]
@LefterisJP would you like to check this, since we discussed it in the morning? For me, everything seems ok but since you had some reservations I will leave it to you.

**Comment by yabirgb** [comment]
So I took a look at this PR and having this 3 steps felt natural. It was similar to experiences in other websites. My only concern is that for me it feels a bit strange to ask whether to restore or not the premium database as first step but after giving a thinking it made sense, not only in the technical side but also in the UX.

We can give it a try and collect feedback (maybe in a twitter poll or in discord) and see what the users think. If not we can iterate over this design.

https://media.discordapp.net/attachments/852835580013838376/922486056350265394/unknown.png?width=841&height=681

**Pull request merged**